### PR TITLE
Feature/error pages

### PR DESCRIPTION
**Pull request by rjweeks**
Create draft and send message warning pages.

### CODE REVIEW

File: frontstage/views/secure_messaging.py [hunk 1]
```diff
@@ -69,15 +69,17 @@ def create_message(session):
         if request.form['submit'] == 'Save draft':
             if "msg_id" in request.form:
                 data['msg_id'] = request.form['msg_id']
+                headers['Authorization'] = request.cookies['authorization']
                 response = requests.put(DRAFT_PUT_API_URL.format(request.form['msg_id']), data=json.dumps(data), headers=headers)
```
Comment (andrewmil) [inline frontstage/views/secure_messaging.py]: for some reason I've been able to hit this line with an empty `msg_id` which send you to endpoint `/draft//modify` which returns a 404

File: frontstage/views/secure_messaging.py [hunk 2]
```diff
@@ -69,15 +69,17 @@ def create_message(session):
...
                 if response.status_code != 200:
-                    # TODO replace with custom error page when available
-                    return redirect(url_for('error_bp.error_page'))
+                    get_json = json.loads(response.content)
+                    return render_template('secure-messages-draft.html', _theme='default', draft=data, errors=get_json)
+
             else:
                 response = requests.post(DRAFT_SAVE_API_URL, data=json.dumps(data), headers=headers)
                 if response.status_code != 201:
```
Comment (andrewmil) [inline frontstage/views/secure_messaging.py]: This probably needs to be more specific, If the api returns a 404, 500 we dont want to be returning the error messages

File: frontstage/views/secure_messaging.py [hunk 3]
```diff
@@ -129,13 +129,13 @@ def reply_message(session):
                 data['msg_id'] = request.form['msg_id']
                 response = requests.put(DRAFT_PUT_API_URL.format(request.form['msg_id']), data=json.dumps(data), headers=headers)
                 if response.status_code != 200:
-                    # TODO replace with custom error page when available
-                    return redirect(url_for('error_bp.error_page'))
+                    get_json = json.loads(response.content)
+                    return render_template('secure-messages-draft.html', _theme='default', draft=get_json, data={"subject" : "Subject field exceeds 100 characters."})
```
Comment (andrewmil) [inline frontstage/views/secure_messaging.py]: should the data here be hardcoded?

**Comment by andrewmil** [comment]
I'm also still getting the errors that darrell references in his review at https://github.com/ONSdigital/ras-secure-message/pull/71

File: frontstage/templates/secure-messages-create.html [hunk 4]
```diff
@@ -15,28 +19,45 @@ <h1 class="saturn">Create message</h1>
...
                     <fieldset>
                         <label for="secure-message-subject">Subject</label>
                         <input
                             id="secure-message-subject"
                             class="secure-message-form__subject input input--text"
                             type="text"
                             name="secure-message-subject"
-                            required
-                            maxlength="100"/>
+                            value="{{draft['subject']}}" />
```
Comment (andrewmil) [inline frontstage/templates/secure-messages-create.html]: `value="{{ draft['subject'] }}"`
missing spaces

File: frontstage/templates/secure-messages-create.html [hunk 5]
```diff
@@ -15,28 +19,45 @@ <h1 class="saturn">Create message</h1>
...
+
                     <fieldset>
                         <label for="secure-message-body">Message</label>
                         <textarea
                             id="secure-message-body"
                             class="secure-message-form__body input"
                             rows="10"
-                            name="secure-message-body"
-                            maxlength="10000"></textarea>
+                            name="secure-message-body">{{ draft ['body'] }}</textarea>
```
Comment (andrewmil) [inline frontstage/templates/secure-messages-create.html]: space between `draft` and `['body']` this might cause issues

**Comment by emilioward** [comment]
I dont get the errors that you mentioned @andrewmil

File: frontstage/templates/secure-messages-create.html [hunk 6]
```diff
@@ -15,28 +19,45 @@ <h1 class="saturn">Create message</h1>
...
                     </fieldset>
+
+                    {% if "body" in errors %}
+                    </div>
+                    {% endif %}
```
Comment (dcdarrell9) [inline frontstage/templates/secure-messages-create.html]: What does this do?

**Comment by andrewmil** [comment]
screen shot 2017-08-08 at 11 37 56
Message not in correct format

File: frontstage/templates/secure-messages-create.html (hunk 6 above)
Comment (andrewmil) [inline frontstage/templates/secure-messages-create.html]: creates a closing div for the error box. It's not very elegant but works for now, the other frontstage form pages do a similar thing

File: frontstage/templates/secure-messages-create.html (hunk 5 above)
Comment (rjweeks) [inline frontstage/templates/secure-messages-create.html]: fixed

File: frontstage/templates/secure-messages-create.html (hunk 4 above)
Comment (rjweeks) [inline frontstage/templates/secure-messages-create.html]: fixed

File: frontstage/views/secure_messaging.py (hunk 2 above)
Comment (rjweeks) [inline frontstage/views/secure_messaging.py]: Let's do more serious refactoring after these changes are merged.

**Pull request merged**